parsers/BR.py

```diff
@@ -1,191 +1,179 @@
-        'flow': -1
-    },
-    'PY': {
-        'name': u'paraguai',
-        'flow': -1
-    }
+GENERATION_MAPPING = {
+    "nuclear": "nuclear",
+    "eolica": "wind",
+    "termica": "unknown",
+    "solar": "solar",
+    "hydro": "hydro",
+}
+
+REGIONS = {
+    "BR-NE": "nordeste",
+    "BR-N": "norte",
+    "BR-CS": "sudesteECentroOeste",
+    "BR-S": "sul",
+}
+
+REGION_EXCHANGES = {
+    "BR-CS->BR-S": "sul_sudeste",
+    "BR-CS->BR-NE": "sudeste_nordeste",
+    "BR-CS->BR-N": "sudeste_norteFic",
+    "BR-N->BR-NE": "norteFic_nordeste",
+}
+
+
+REGION_EXCHANGES_DIRECTIONS = {
+    "BR-CS->BR-S": -1,
+    "BR-CS->BR-NE": 1,
+    "BR-CS->BR-N": 1,
+    "BR-N->BR-NE": 1,
+}
+
+COUNTRIES_EXCHANGE = {
+    "UY": {"name": "uruguai", "flow": 1},
+    "AR": {"name": "argentina", "flow": -1},
+    "PY": {"name": "paraguai", "flow": -1},
 }
 
 
 def get_data(session, logger):
     """Requests generation data in json format."""
-
     s = session or requests.session()
-    json_data = s.get(url).json()
+    json_data = s.get(URL).json()
+
     return json_data
 
 
 def production_processor(json_data, zone_key) -> tuple:
     """Extracts data timestamp and sums regional data into totals by key."""
 
-    dt = arrow.get(json_data['Data'])
+    dt = arrow.get(json_data["Data"])
     totals = defaultdict(lambda: 0.0)
 
-    region = regions[zone_key]
-    breakdown = json_data[region][u'geracao']
+    region = REGIONS[zone_key]
+    breakdown = json_data[region]["geracao"]
     for generation, val in breakdown.items():
         totals[generation] += val
 
     # BR_CS contains the Itaipu Dam.
     # We merge the hydro keys into one, then remove unnecessary keys.
-    totals['hydro'] = totals.get(u'hidraulica', 0.0) + totals.get(u'itaipu50HzBrasil', 0.0) + totals.get(u'itaipu60Hz', 0.0)
-    entriesToRemove = (u'hidraulica', u'itaipu50HzBrasil', u'itaipu60Hz', u'total')
-    for k in entriesToRemove:
-        totals.pop(k, None)
-
-    mapped_totals = {generation_mapping.get(name, 'unknown'): val for name, val
-                     in totals.items()}
+    totals["hydro"] = (
+        totals.get("hidraulica", 0.0)
+        + totals.get("itaipu50HzBrasil", 0.0)
+        + totals.get("itaipu60Hz", 0.0)
+    )
+    entries_to_remove = {"hidraulica", "itaipu50HzBrasil", "itaipu60Hz", "total"}
+    mapped_totals = {
+        GENERATION_MAPPING.get(name, "unknown"): val
+        for name, val in totals.items()
+        if name not in entries_to_remove
+    }
 
     return dt, mapped_totals
 
 
 def fetch_production(zone_key, session=None, target_datetime=None, logger=None) -> dict:
     """Requests the last known production mix (in MW) of a given country."""
     if target_datetime:
-        raise NotImplementedError('This parser is not yet able to parse past dates')
+        raise NotImplementedError("This parser is not yet able to parse past dates")
 
     gd = get_data(session, logger)
     generation = production_processor(gd, zone_key)
```

Comment:
Oh, one more thing. This could be a little nicer as:

```suggestion
    timestamp, production = production_processor(gd, zone_key)
```

Then below

```python
        "datetime": timestamp.datetime,
        "production": production,
```

Rather than using somewhat nondescript and mysterious `generation[0]` and `generation[1]`